File: week07/19/P3.py
def solution(A):
    B = sorted(A)
    if (B[-1] * B[-2] * B[-3] > B[-1] * B[0] * B[1]):
        return B[-1] * B[-2] * B[-3]
    return B[-1] * B[0] * B[1]

if __name__ == "__main__":
    A = [-7, 1, 2, -4, 5, 6]
    print(solution(A))

File: week07/19/P3.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

bool compare(int a, int b) {
    return a > b;
}

class Solution {
public:
    int solution(vector<int> A) {
        sort(A.begin(), A.end(), compare);
        if (A[0] * A[1] * A[2] > A[0] * A[A.size()-1] * A[A.size()-2]) {
            return A[0] * A[1] * A[2];
        }
        return A[0] * A[A.size()-1] * A[A.size()-2];
    }
};

int main() {
    vector<int> A = {-7, 1, 2, -4, 5, 6};

    Solution result;
    cout << result.solution(A) << endl;

    return 0;
}